server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -529,7 +531,7 @@
     private Map<Long, VmAndCountDetails> vmIdCountMap = new ConcurrentHashMap<>();
 
     private static final int MAX_HTTP_GET_LENGTH = 2 * MAX_USER_DATA_LENGTH_BYTES;
-    private static final int MAX_HTTP_POST_LENGTH = 16 * MAX_USER_DATA_LENGTH_BYTES;
+    private static final int MAX_HTTP_POST_LENGTH = 512 * MAX_USER_DATA_LENGTH_BYTES;
```

Comment:
why should this change if we make it a global setting?

Reply:
@DaanHoogland Default is 32K and max is 1MB
So using the global setting you can change values and this prevents setting the value above 1MB

Reply:
What I don't understand is the change from 16 to 512, @ravening why is that? Can you name a contant for 512 (former 16)?

Reply:
@DaanHoogland earlier, the max length of post request is 32K. so 16 * 2K. Now it can support upto 1MB. So 512 * 2K

Reply:
```suggestion
    private static final int NUM_OF_2K_BLOCKS = 512;
    private static final int MAX_HTTP_POST_LENGTH = NUM_OF_2K_BLOCKS * MAX_USER_DATA_LENGTH_BYTES;
```